### PR TITLE
Testcases

### PR DESCRIPTION
**Pull request by byteZorvin**
- Added basic testcases
- Made few code improvements
- Added consume withdrawal quota in `token_bridge.cairo`
- Changed events visibility

### CODE REVIEW

File: src/bridge/token_bridge.cairo [hunk 1]
```diff
@@ -86,12 +86,13 @@ pub mod TokenBridge {
...
         pub const CANNOT_BLOCK: felt252 = 'Cannot block';
         pub const INVALID_RECIPIENT: felt252 = 'Invalid recipient';
+        pub const MAX_BALANCE_EXCEEDED: felt252 = 'Max Balance Exceeded';
     }
 
 
     #[derive(Drop, starknet::Event)]
     #[event]
-    enum Event {
+    pub enum Event {
```
[the remark this answers is not in the document]
Comment (byteZorvin) [reply src/bridge/token_bridge.cairo:95]: IMO making events public makes sense, if there is something else we can rethink

File: src/bridge/token_bridge.cairo [hunk 2]
```diff
@@ -460,16 +465,19 @@ pub mod TokenBridge {
...
+
+            let old_settings = self.token_settings.read(token);
+            let new_settings = TokenSettings {
                 token_status: TokenStatus::Pending,
                 deployment_message_hash: deployment_message_hash,
                 pending_deployment_expiration: get_block_timestamp()
                     + constants::MAX_PENDING_DURATION.try_into().unwrap(),
-                max_total_balance: core::integer::BoundedInt::max(),
-                withdrawal_limit_applied: false
+                ..old_settings
```
[the remark this answers is not in the document]
Comment (byteZorvin) [reply src/bridge/token_bridge.cairo:477]: You can block and set max balance for a token even before its is activated

File: tests/token_bridge_test.cairo [hunk 3]
```diff
@@ -1,35 +1,27 @@
...
-        let mut calldata = ArrayTrait::new();
-        l3_bridge_address.serialize(ref calldata);
-        appchainContract.serialize(ref calldata);
-        let (token_bridge_address, _) = token_bridgeCH.deploy(@calldata).unwrap();
-
-        (ITokenBridgeDispatcher { contract_address: token_bridge_address }, token_bridge_address)
-    }
 
     fn deploy_erc20(name: ByteArray, symbol: ByteArray) -> ContractAddress {
         let recipient: felt252 = 'owner'.try_into().unwrap();
```
[the remark this answers is not in the document]
Comment (byteZorvin) [reply tests/token_bridge_test.cairo:27]: Yes makes sense, changed it

File: src/bridge/token_bridge.cairo [hunk 4]
```diff
@@ -460,16 +465,19 @@ pub mod TokenBridge {
                 .sn_to_appchain_messages(deployment_message_hash);
             assert(nonce.is_non_zero(), Errors::DEPLOYMENT_MESSAGE_DOES_NOT_EXIST);
 
-            let token_status = TokenSettings {
+            // Reading existing settings as withdrawal_limit_applied and max_total_balance
```
Comment (byteZorvin) [inline src/bridge/token_bridge.cairo:468]: The comment mentiones the need to initialise settings from previous settings

File: src/bridge/token_bridge.cairo [hunk 5]
```diff
@@ -576,10 +583,11 @@ pub mod TokenBridge {
             assert(recipient.is_non_zero(), Errors::INVALID_RECIPIENT);
 
             self.consume_message(token, amount, recipient);
-            let settings = self.token_settings.read(token);
-            // TODO: Consume quota from here
-            // DEP(byteZorvin): Complete the withdrawal component in cairo 
-            if (settings.withdrawal_limit_applied) {}
+
+            if (self.is_withdrawal_limit_applied(token)) {
```
[the remark this answers is not in the document]
Comment (byteZorvin) [reply src/bridge/token_bridge.cairo:587]: Yes. Removing it

File: src/bridge/token_bridge.cairo (hunk 2 above)
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply src/bridge/token_bridge.cairo:477]: oh ok

**Pull request merged**